File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.php
<?php
class Solution {
	/*
	 @param String $s
 	@return Integer
	*/
	function lengthOfLongestSubstring($s) {
		$head = 0; $tail = 0; $set = array (); $count = 0; $high = 0;
		while ($head < strlen($s)) {
			while ($set[$s[$head]] == 1) {
				$set[$s[$tail++]]--;
			}
			$set[$s[$head++]] = 1;
			$count = ($head - $tail);
			if ($count > $high) $high = $count;
		}
		return $high;
	}
}

?>

File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.c
#include <stdio.h>

int lengthOfLongestSubstring(char * s){
	int alpha[128] = {0};
	int sol = 0;
	int head = 0;
	int tail = 0;
	int len = 0;
	while (s[head]!='\0') {
		while (alpha[s[head]] == 1) {
			alpha[s[tail++]]--;
		}
		alpha[s[head++]]++;
		len = head-tail;
		if (len > sol) sol = len;
	}
	return sol;
}